xmake/core/tool/builder.lua

```diff
@@ -708,7 +708,9 @@ function builder:_preprocess_flags(flags)
             local flagkey = type(flag) == "table" and table.concat(flag, "") or flag
             if flag and not unique[flagkey] then
                 table.insert(flags_new, flag)
-                unique[flagkey] = true
+                if flagkey:startswith("-l") or flagkey:startswith("-L") or flagkey:startswith("/l") or flagkey:startswith("/L") then
+                    unique[flagkey] = true
+                end
```

Comment:
we need to deduplicate more flags, e.g. -Dxx ...

So we shouldn't do this, hard coding is not universal.